pkg/scalers/splunk_observability_scaler.go

```diff
@@ -0,0 +1,183 @@
+
+	if err := config.TypedConfig(meta); err != nil {
+		return nil, fmt.Errorf("error parsing splunk observability metadata: %w", err)
+	}
+
+	return meta, nil
+}
+
+func newSplunkO11yConnection(meta *splunkObservabilityMetadata, logger logr.Logger) (*signalflow.Client, error) {
+	logger.Info(fmt.Sprintf("meta: %+v\n", meta))
+
+	if meta.Realm == "" || meta.AccessToken == "" {
+		return nil, fmt.Errorf("error: Could not find splunk access token or ream")
+	}
+
+	apiClient, err := signalflow.NewClient(
+		signalflow.StreamURLForRealm(meta.Realm),
+		signalflow.AccessToken(meta.AccessToken),
+		signalflow.OnError(func(err error) {
+			errorMsg := fmt.Sprintf("error in SignalFlow client: %v\n", err)
+			logger.Info(errorMsg)
+		}))
+	if err != nil {
+		return nil, fmt.Errorf("error creating SignalFlow client: %w", err)
+	}
+
+	return apiClient, nil
+}
+
+func NewSplunkObservabilityScaler(config *scalersconfig.ScalerConfig) (Scaler, error) {
+	logger := InitializeLogger(config, "splunk_observability_scaler")
+
+	meta, err := parseSplunkObservabilityMetadata(config)
+	if err != nil {
+		return nil, fmt.Errorf("error parsing Splunk metadata: %w", err)
+	}
+
+	apiClient, err := newSplunkO11yConnection(meta, logger)
+	if err != nil {
+		return nil, fmt.Errorf("error establishing Splunk Observability Cloud connection: %w", err)
+	}
+
+	return &splunkObservabilityScaler{
+		metadata:  meta,
+		apiClient: apiClient,
+		logger:    logger,
+	}, nil
+}
+
+func (s *splunkObservabilityScaler) getQueryResult() (float64, error) {
+	comp, err := s.apiClient.Execute(context.Background(), &signalflow.ExecuteRequest{
+		Program: s.metadata.Query,
+	})
+	if err != nil {
+		return -1, fmt.Errorf("error: could not execute signalflow query: %w", err)
+	}
+
+	s.logger.Info("Started MTS stream.")
+
+	time.Sleep(time.Duration(s.metadata.Duration * int(time.Second)))
+	if err := comp.Stop(context.Background()); err != nil {
+		return -1, fmt.Errorf("error creating SignalFlow client: %w", err)
+	}
+
+	s.logger.Info("Closed MTS stream.")
+
+	max := math.Inf(-1)
+	min := math.Inf(1)
+	valueSum := 0.0
+	valueCount := 0
+	s.logger.Info("Now iterating over results.")
+	for msg := range comp.Data() {
+		if len(msg.Payloads) == 0 {
+			s.logger.Info("No data retreived.")
+			continue
+		}
+		for _, pl := range msg.Payloads {
+			value, ok := pl.Value().(float64)
+			if !ok {
+				return -1, fmt.Errorf("error: could not convert Splunk Observability metric value to float64")
+			}
+			s.logger.Info(fmt.Sprintf("Encountering value %.4f\n", value))
+			max = math.Max(max, value)
+			min = math.Min(min, value)
+			valueSum += value
+			valueCount++
+		}
+	}
+
+	if valueCount > 1 && s.metadata.QueryAggregator == "" {
+		return 0, fmt.Errorf("query returned more than 1 series; modify the query to return only 1 series or add a queryAggregator")
+	}
+
+	switch s.metadata.QueryAggregator {
+	case "max":
+		s.logger.Info(fmt.Sprintf("Returning max value: %.4f\n", max))
```

Comment:
I don't think we need to log every value being returned. The function should just execute the logic. Otherwise, the logs will get pretty noisy

Reply:
It was just for me as a sort of poor-mans-debugging. Let me get back to you with this.

Reply:
changed "s.logger.Info" to "s.logger.V(1).Info". This means the log only shows if debugging is enabled.